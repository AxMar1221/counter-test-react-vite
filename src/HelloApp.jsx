// function App(){
//     return ( <h1>Hi Tachi</h1>);
// }
// export default App;

export const Helloapp = () => {
  return (
    <h1>Helloapp</h1>
  )
}
