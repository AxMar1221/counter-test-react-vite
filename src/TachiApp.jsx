import PropTypes from 'prop-types'

// const getSuma = (a, b) => {
//   return a + b;
// }

// const newValue = {
//   message: 'Tahci',
//   tittle: 'Developer JR'
// };

// export const TachiApp = ( props ) => {
export const TachiApp = ( { title, subTitle, name } ) => {

  // console.log(props);
   
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      {/* <h1> { props.title } </h1> */}
      {/* <h1> { title } </h1>
      <h1> { getSuma(21,12)} </h1>
      <code> { JSON.stringify( newValue  ) } </code>
      <p>Soy Developer Jr</p> */}
      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  )
}

TachiApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string
}

TachiApp.defaultProps = {
  name: 'Tachi Dito',
  subTitle: 'No hay subtitulo',
  title: 'No hay titulo',
}