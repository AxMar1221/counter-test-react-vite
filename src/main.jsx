import React from 'react';
import ReactDOM  from 'react-dom/client';
// import { CounterApp } from './CounterApp';
import { TachiApp } from './TachiApp';
// import { Helloapp } from './Helloapp';
import './styles.css'


ReactDOM.createRoot( document.getElementById('root')).render(
    <React.StrictMode>
        {/* <CounterApp value={ 21 }/> */}
        <TachiApp title={'Hola soy Tachi'}/>
    </React.StrictMode>
);