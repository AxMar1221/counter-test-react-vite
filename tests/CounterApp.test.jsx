import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe('Pruebas en componente CounterApp', () => {

    const valueInitial = 21;

    test('debeb de hacer mach con el snapshot', () => {
      const { container } = render( <CounterApp value={ valueInitial }/>)
      expect( container ).toMatchSnapshot();
    });

    test('debe de mostrar el valor unicial de 12 en CounterApp', () => {
      render( <CounterApp value={ 12 } />);
      expect( screen.getByText(12) ).toBeTruthy();
    });

    test('debe de incrementar con el boton +1', () => {
        render( <CounterApp value={ valueInitial } />);
        fireEvent.click( screen.getByText('+1'))
        expect( screen.getByText('22')).toBeTruthy();
    });
    test('debe de decrementar con el boton -1', () => {
        render( <CounterApp value={ valueInitial } />);
        fireEvent.click( screen.getByText('-1'))
        expect( screen.getByText('20')).toBeTruthy();
    });
    test('debeb de funcionar el reset', () => {
        render( <CounterApp value={ 12 } />);
        fireEvent.click( screen.getByText('-1'))
        fireEvent.click( screen.getByText('+1'))
        fireEvent.click( screen.getByText('-1'))
        // fireEvent.click( screen.getByText('Reset'))
        fireEvent.click( screen.getByRole('button', {name:'btn-reset'}))
        expect( screen.getByText( 12 )).toBeTruthy();
    })
    
})
