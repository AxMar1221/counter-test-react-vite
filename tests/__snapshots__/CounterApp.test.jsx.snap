// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente CounterApp debeb de hacer mach con el snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
     
    21
     
  </h2>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;
